src/tools/cephfs/cephfs-shell

```diff
@@ -1040,25 +1040,54 @@ sub-directories, files')
         """
         self.poutput(os.getcwd())
 
+    def complete_df(self, text, line, begidx, endidx):
+        """
+        auto complete of file name.
+        """
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    df_parser = argparse.ArgumentParser(
+        description='Show information about the file system on which each File resides,\
+            or all file systems by default.')
+    df_parser.add_argument('file', help='Name of the file', type=str,
+                           nargs='*', default='')
+
+    @with_argparser(df_parser)
     def do_df(self, arglist):
         """
         Display the amount of available disk space for file systems
         """
-        for index, i in enumerate(list_items(cephfs.getcwd())[2:]):
-            if index == 0:
-                self.poutput('{:25s}\t{:5s}\t{:15s}{:10s}{}'.format(
-                    "1K-blocks", "Used", "Available", "Use%", "Stored on"))
-            if not is_dir_exists(i.d_name):
-                statfs = cephfs.statfs(i.d_name)
-                stat = cephfs.stat(i.d_name)
+        flist = []  # List of valid existing directory or file names
+
+        for files in arglist.file:
+            if is_dir_exists(files) or is_file_exists(files):
+                flist.append(files)
+            else:
+                print("df: {}: No such file or directory".format(files))
```

Comment:
Using the `try/except` below, we can delete this above loop.

Reply:
With this loop the valid file and invalid file output are kept separate. Otherwise they would be mixed.
`CephFS:~/>>> df q t y`
`df: q: No such file or directory`
`df: y: No such file or directory`
`1K-blocks     Used 	Available      Use%      Stored on`
`9318400	  0	 9318400	       0%       t`
Can we keep this loop?

Reply:
As long as the errors go to stderr, this is normal behavior. For example:

````
pdonnell@icewind ~$ echo hello > bar
pdonnell@icewind ~$ cat foo bar rab
/bin/cat: foo: No such file or directory
hello
/bin/cat: rab: No such file or directory
````

Reply:
Thank you for explaining. I understand it but the normal behaviour for df command is different.
I just wanted to make the output similar to unix shell. The output looks clean if valid and invalid file output are separate.
```
[root@localhost build]# df dev i out
df: i: No such file or directory
Filesystem                        1K-blocks     Used Available Use% Mounted on
/dev/mapper/fedora_dhcp35--164-02 125869412 34577512  84855036  29% /home
/dev/mapper/fedora_dhcp35--164-02 125869412 34577512  84855036  29% /home
```
If you still prefer to have the valid and invalid file outputs together. I will modify it.

Reply:
They only appear together when displayed on the terminal. If you redirect stderr to /dev/null, then you don't see it. I prefer to mix them in this way.